### ISSUE TITLE
No nullable warning emmited for non-nullable properties on structs. 

### ISSUE THREAD

**Issue by KimBrusevold** [dotnet/roslyn#74715]
**Version Used**: 
Using SDK .Net 8.0.107, x64

**Steps to Reproduce**:
https://sharplab.io/#v2:CYLg1APgAgTAjAWAFBQMwAJboMLoN7LpGYZQAs6AsgBQCU+hxTAbgIYBO6ALgKYDOXdAF50AOx4B3dAGUu7AK4BjLgBV+XOgG5GTIlDgBOarwEA6NQA8u5gPYBVAA4Oe7OrW1Idutp15c+AELCYpI4ADasfHxqAlrIXkz6Rn6B5jxWto7OrrTuXgC+yIWeJfEoGAIKyjJySqrqyARIiaRwAAzoloJ46ADmPFya6HwDQ8XFyGiYMOGR0Q1ITS2Y7Z3p3X2jw1vjyEA===

Enable nullable in project file 
` <Nullable>enable</Nullable>`

Create one struct and one class:

```csharp

struct StructTest
{
    public string Text { get; set; }
}

class ClassTest
{
    public string Text { get; set; }
}
```

**Expected Behavior**:

A nullable warning that the non-nullable property 'Text' cannot be initialized to null on both the struct and class.

**Actual Behavior**:

Only the class property emits the error:
"Non-nullable property 'Text' must contain a non-null value when exiting constructor. Consider adding the 'required' modifier or declaring the property as nullable.[CS8618](https://msdn.microsoft.com/query/roslyn.query?appId%3Droslyn%26k%3Dk%28CS8618%29)"

If we treat this warning as an error, the compiler generates invalid code. By obtaining a reference to the struct property Text, we may get an null reference error, even though we declared the property as non-nullable. The empty initializer for the struct (new StructTest()) which calls default(StructTest) sets null in the non-nullable fields

**Comment by RikkiGibson** [comment]
Unfortunately, structs have known/expected gaps with null safety. There have been various proposals to address this over the years such as dotnet/csharplang#146.

But, the struct scenario you have shown is not expected to issue any warnings under the current design. One way to work around this and get a certain (limited) degree of nullability checking is to explicitly declare a constructor in `struct StructTest`. There will still be null safety gaps when the default value is used, however.

**Comment by KimBrusevold** [comment]
Thanks. I'm not at liberty to say anything about what is expected and not. 

I got around this by creating properties and marking them as required. The compiler recognized that de default "empty" constructor is invalid (which is a good thing). 
The nullable feature is such a good thing, and seems so close to being good enough. 

I really want to be confident in the compiler checking the nullable stuff for me.
